tests/unit/test_repo_manager.py

```diff
@@ -0,0 +1,180 @@
+"""
+Tests for the repo manager tool
+"""
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import os
+import tempfile
+import unittest
+
+import ga4gh.repo_manager as repo_manager
+import ga4gh.exceptions as exceptions
+
+
+class AbstractRepoManagerTest(unittest.TestCase):
+
+    vcfDirPath = os.path.abspath(
+        'tests/data/datasets/dataset1/variants/1kgPhase1')
+    vcfDirPath2 = os.path.abspath(
+        'tests/data/datasets/dataset1/variants/1kgPhase3')
+    faPath = os.path.abspath(
+        'tests/data/referenceSets/Default/chr17.fa.gz')
+    faPath2 = os.path.abspath(
+        'tests/data/referenceSets/example_1/simple.fa.gz')
+    faPath3 = os.path.abspath(
+        'tests/data/referenceSets/example_2/random1.fa.gz')
+    bamPath = os.path.abspath(
+        'tests/data/datasets/dataset1/reads/chr17.1-250.bam')
+    bamPath2 = os.path.abspath(
+        'tests/data/datasets/dataset1/reads/'
+        'wgEncodeUwRepliSeqBg02esG1bAlnRep1_sample.bam')
+    moveMode = 'link'
+
+
+class RepoManagerTest(AbstractRepoManagerTest):
+    """
+    End to end test for the repo manager
+    """
+    def testEndtoEnd(self):
+        # TODO can't use tempdir here since its contents are destroyed
+        # before the finally block executes ... not sure why
+        testDirName = 'repo-manager-test-dir'
```

Comment:
Use setUp and tearDown here to create and delete ``self.testDirName`` using the tempfile library.